### PR TITLE
Fix/refactor activity class

### PR DESCRIPTION
**Pull request by Shawntru**
## Is this a fix or a feature?
Fix
## What is the change?
Activity.js
## What does it fix?
Refactor methods for activity class down to getActiveMinutesByDate()
## Where should the reviewer start?
activity.js line 1
## How should this be tested?
Use built in chai mocks tests in activity-test.js

@Shawntru 
@adavidson02
@sarah01123

### CODE REVIEW

**Review by ADavidson02 — APPROVED** [review]
Looks good

File: src/Activity.js [hunk 1]
```diff
@@ -1,50 +1,95 @@
...
+      return totalMinutes;
+    }, 0);
+    return (Math.round((totalTime / 7) * 10) / 10);
   }
 
-  accomplishStepGoal(id, date, userRepo) {
-    const userStepsByDate = this.activityData.find((data) => id === data.userID && date === data.date);
-    if (userStepsByDate.numSteps === userRepo.dailyStepGoal) {
-      return true;
+  findGivenWeek(date) {
```
Comment (saraho1123) [inline src/Activity.js:42]: This is great!

File: src/Activity.js [hunk 2]
```diff
@@ -1,50 +1,95 @@
...
 
-  calculateActiveAverageForWeek(id, date, userRepo) {
-    return parseFloat((userRepo.getWeekFromDate(date, id, this.activityData).reduce((acc, elem) => acc += elem.minutesActive, 0) / 7).toFixed(1));
+  calculateActiveAverageForWeek(date) {
+    let week = this.findGivenWeek(date);
+    let totalTime = week.reduce((totalMinutes, element) => {
+      totalMinutes += element.minutesActive;
+      return totalMinutes;
+    }, 0);
+    return (Math.round((totalTime / 7) * 10) / 10);
```
Comment (saraho1123) [inline src/Activity.js:39]: I'm having trouble wrapping my head around multiplying by 10 and then dividing by 10. Gonna look at this one more so I understand it better!

**Review by saraho1123 — APPROVED** [review]

File: src/Activity.js (hunk 2 above)
Comment (Shawntru) [reply src/Activity.js:39]: Useful for rounding!

Comment (saraho1123) [reply src/Activity.js:39]: Got a great explanation from @Shawntru Thanks! It is so that the number is rounded to the nearest 10th! Sweet!

**Comment by saraho1123** [comment]
We plan to move many of the unfactor methods to UserRepo class.

**Pull request merged**